### DISCUSSION TITLE
TypeError: Cannot read properties of undefined (reading 'replaceAll')

### DISCUSSION THREAD

**Discussion by yerassyl-sailaubay** [FlowiseAI/Flowise#420] [Q&A]
I've made this chatbot for an website. Everything waas working fine. But after restarting this issue appeared 
![image](https://github.com/FlowiseAI/Flowise/assets/107134360/c7f85a97-4ca1-4714-a98b-004ff9ced4ff)

**Comment by VikingDadMedic** [comment]
You are encountering a problem akin to this issue - #267

Trying to explain "why" will be overly complex - so let me just explain the solution (and you should be able to connect the dots 🙃). 

For future reference, you're correct in your assumption that it's the restart that triggers the issue. Though technically it really isn't an issue in the strictest since. 

Assuming you initially started your Pinecone index empty, if you navigate to Pinecone dashboard you should see that there are still vectors in there from prior to you restarting. 

Once confirmed that you upserted vectors are there, you can go back to Flowise. Delete the splitter, and scraper nodes

Next, swap out the **Pinecone Upsert Document** node and replace with the  **Pinecone Load Existing Index** node (_don't forget to update the new node's parameter if you are using a named space in your index - or want to filter by metadata if you included any in your scrape_). 

And finally give the whole thing a Save & Refresh - et voila! 

See below for attached visuals.

**Before**
![before](https://github.com/FlowiseAI/Flowise/assets/91684138/6f7eb135-65bc-4c55-bc75-191825c60a3b)

**After**
![after](https://github.com/FlowiseAI/Flowise/assets/91684138/42de7f7c-c660-4b6f-8c5e-1d382f001904)

**Reply by klarak777** [reply]
Thanks, it works.

**Reply by odysseyalive** [reply]
I'm curious, why does deleting a node and adding it back end "fix the issue"?

For instance, I can delete a node that I was having issues with. If I don't add it back end, It kills anything I try to do in the current session and any other sessions. (see example image where it's looking for the missing tool).

Where are these history's being saved to? Do I need to reset langchain?

You mentioned the process is complicated, what am I searching for in the documentation to find out more?

![What do I reset here?](https://i.imgur.com/vxG0fWn.png)

**Comment by dabrodev** [comment]
I have the same issue but for me, it doesn't work from the beginning. The index is empty.

**Comment by elyesj23** [comment]
Had the same error, but fixed it by just deleting it and rebuilding again

**Comment by trylof** [comment]
I'm having the same issue, including an empty index and deleting and rebuilding doesn't help

**Reply by rcraw** [reply]
Me too. built from scratch, failed on the first upsert attempt. Pinecone index is empty, same error.

**Reply by rcraw** [reply]
just deleted and brought back in the pincone vectore store, worked fine after that.
